quantities/named_quantities.hpp

```diff
@@ -23,8 +24,23 @@ template<typename Q>
 using Inverse = Quotient<double, Q>;
 
 template<typename T, int exponent>
-using Exponentiation =
-    typename _generators::ExponentiationGenerator<T, exponent>::Type;
+struct ExponentiationGenerator : not_constructible {
```

Comment:
Having to define a generator in this header is polluting it and breaking the separation of concern between the files.

But it's easy to fix I think: just move it back to `generators.hpp`, `_body.hpp` and pass `Product` and `Quotient` as two template template parameters.

It makes a lot of sense conceptually too: in order to define exponentiation, you need to tell me how you multiply and divide.

Reply:
It is not that kind of generator anymore; it is more like DerivativesGenerator.

Reply:
At least, hide it in the body.  We don't put the implementation in the headers.

Also, what is the final goal?  Will this move to another file?  To another project?

Reply:
> what is the final goal? Will this move to another file? To another project?

Maybe. (Like Product, it is not actually quantities-specific.) But not today. At least it can move now.

Reply:
As discussed, we cannot hide this in a body without moving it. I will move it in a subsequent pull request.